### PR TITLE
validations and channel endpoint

### PR DESCRIPTION
**Pull request by yartman**
This pr is for:
* adding topics to channels
* subscriber and publisher role validations
* GET available channels from MessageBroker
* fqcn validation on channel creation

### CODE REVIEW

File: apps/dsb-message-broker/src/transport/transport.module.ts [hunk 1]
```diff
@@ -15,6 +15,10 @@ const transportFactory = {
         const privateKey = configService.get('PRIVATE_KEY');
         const mbDID = configService.get('MB_DID');
 
+        // console.log("mbDID ", mbDID);
```
Comment (adamstaveley) [inline apps/dsb-message-broker/src/transport/transport.module.ts:18]: remove

File: apps/dsb-message-broker/src/message/error.handler.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+export const messageErrorHandler = (error: any) => {
+    if (error instanceof ChannelNotFoundError) {
+        throw new BadRequestException({ message: error.message });
+    }
+    if (error instanceof TransportUnavailableError) {
+        throw new ServiceUnavailableException();
+    }
+
+    if (
+        error.message === 'Not authurized to publish' ||
```
Comment (adamstaveley) [inline apps/dsb-message-broker/src/message/error.handler.ts:17]: typo: authorized

File: apps/dsb-message-broker/src/transport/transport.module.ts (hunk 1 above)
Comment (yartman) [reply apps/dsb-message-broker/src/transport/transport.module.ts]: removed

File: apps/dsb-message-broker/src/message/error.handler.ts (hunk 2 above)
Comment (yartman) [reply apps/dsb-message-broker/src/message/error.handler.ts]: edited

**Review by adamstaveley — APPROVED** [review]

**Pull request merged**